embedded-graphics/src/primitives/polyline/styled.rs

```diff
@@ -72,19 +116,62 @@ where
     where
         D: DrawTarget<Color = C>,
     {
-        display.draw_iter(self.into_pixels())
+        if let Some(stroke_color) = self.style.stroke_color {
+            if self.style.is_transparent() {
+                return Ok(());
+            }
+
+            match self.style.stroke_width {
+                0 => Ok(()),
+                1 => display.draw_iter(
+                    self.primitive
+                        .points()
+                        .map(|point| Pixel(point, stroke_color)),
+                ),
+                _ => {
+                    for line in ScanlineIterator::new(self) {
+                        display.fill_solid(
+                            &Rectangle::with_corners(line.start, line.end),
+                            stroke_color,
+                        )?;
+                    }
+
+                    Ok(())
+                }
+            }
+        } else {
+            Ok(())
+        }
     }
 }
 
 impl<C> Dimensions for Styled<Polyline<'_>, PrimitiveStyle<C>>
 where
     C: PixelColor,
 {
-    // NOTE: Polyline currently ignores stroke width, so this delegates to the un-styled bounding
-    // box impl.
     fn bounding_box(&self) -> Rectangle {
```

Comment:
I think this will need to explicitly handle polylines with fewer than two vertices. Otherwise it will return a maximum size rectangle instead of a zero size one.

Reply:
Well spotted. I've added a test case for two and one point with some fixes.